google/cloud/forseti/services/inventory/storage.py

```diff
@@ -154,6 +155,75 @@ def set_error(self, session, message):
         session.add(self)
         session.flush()
 
+    def get_lifecycle_state_details(self, session, resource_type_input):
+        """Count of lifecycle states of the specified resources.
+
+        Generate/return the count of lifecycle states (ACTIVE, DELETE_PENDING)
+        of the specific resource type input (project, folder) for this inventory
+        index.
+
+        Args:
+            session (object) : session object to work on.
+            resource_type_input (str) : resource type to get lifecycle states for.
+
+        Returns:
+            dict: a (lifecycle state -> count) dictionary
+        """
+        resource_data = Inventory.resource_data
+
+        details = dict(
+            session.query(func.json_extract(resource_data, '$.lifecycleState'), func.count())
+            .filter(Inventory.inventory_index_id == self.id)
+            .filter(Inventory.category == 'resource')
+            .filter(Inventory.resource_type == resource_type_input)
+            .group_by(func.json_extract(resource_data, '$.lifecycleState')).all())
+
+        for key in details.keys():
+            new_key = key.replace('\"', '').replace('_', ' ')
+            new_key = ' - '.join([resource_type_input, new_key])
+            details[new_key] = details.pop(key)
+
+        if len(details) == 1:
+            delete_pending_key = ' - '. join([resource_type_input, 'DELETE PENDING'])
```

Comment:
Is it possible to only have 'DELETE PENDING' in details?

Reply:
Correct- fixed.